openassessment/xblock/openassessmentblock.py

```diff
@@ -336,11 +336,22 @@ def file_upload_type(self, value):
         """
         self.file_upload_type_raw = value
 
+    def get_allowed_file_types_or_preset(self):
+        if self.white_listed_file_types:
+            return self.white_listed_file_types
+        elif self.file_upload_type == 'image':
+            return self.ALLOWED_IMAGE_EXTENSIONS
+        elif self.file_upload_type == 'pdf-and-image':
+            return self.ALLOWED_FILE_EXTENSIONS
+        return None
+
     @property
     def white_listed_file_types_string(self):
         """
-        Join the white listed file types into comma delimited string
+        Join the allowed file types into comma delimited string
         """
+        self.white_listed_file_types = self.get_allowed_file_types_or_preset()
```

Comment:
I am not sure if this line is OK, since we have an explicit setter for `white_listed_file_types_string` below.

Reply:
I was debating this piece. By default, it's a no-op (will adopt existing settings for `None` or `custom`), but ratchets forward by loading the preset on first fetch. Maybe the getter is the wrong place to do it but was the way I could think of that introduced the fewest changes.

Reply:
What about creating a constructor and initializing that property there?

Reply:
The XBlock instructions say not to override the constructor function and I can't locate any other common entry point. 

Because of this, we'd end up having to use this preset logic in:
- Course import (loading from XML)
- Student view
- Staff view

We can hook each of these independently, but at the cost of a larger set of changes.